qmxgraph/widget.py

```diff
@@ -465,52 +488,17 @@ def _on_drop(self, event):
         else:
             event.ignore()
 
-    def _set_double_click_bridge(self):
-        """
-        Redirects double click events fired by graph on JavaScript code to
-        Python/Qt side by using a private bridge.
-        """
-        if self.is_loaded():
-            bridge = self._double_click_bridge
-            self._web_view.add_to_js_window(
-                'bridge_double_click_handler', bridge)
-            self.api.set_double_click_handler(
-                'bridge_double_click_handler.on_double_click')
 
-    def _set_popup_menu_bridge(self):
-        """
-        Redirects popup menu (i.e. right click) events fired by graph on
-        JavaScript code to Python/Qt side by using a private bridge.
-        """
-        if self.is_loaded():
-            bridge = self._popup_menu_bridge
-            self._web_view.add_to_js_window(
-                'bridge_popup_menu_handler', bridge)
-            self.api.set_popup_menu_handler(
-                'bridge_popup_menu_handler.on_popup_menu')
+def _make_async_pyqt_slot(*signature, name):
+    import re
 
-    def _set_private_bridge_handler(self, bridge_signal, handler, setter):
-        """
-        Helper method to set handler for private bridges like the ones use for
-        double click and popup menu events.
+    def async_slot(self, *args):
+        signal = getattr(self, signal_name)
+        QTimer.singleShot(1, lambda: signal.emit(*args))
```

Comment:
```suggestion
        QTimer.singleShot(0, lambda: signal.emit(*args))
```

`0` means "schedule for the next event loop"

Reply:
Yes.
But on tests that was causing it to deadlock on "api" calls inside those signals (the situation this async indirection is supposed to avoid).
I can put it is a constant and mock on tests. But even this way I will be afraid it deadlock on production.

Reply:
> But on tests that was causing it to deadlock on "api" calls inside those signals (the situation this async indirection is supposed to avoid).

That doesn't make sense, unless `0` doesn't mean what we think it means. Can you check it again plase?

Reply:
``

Reply:
OK, still don't understand why that's the case, but I don't want to spend more time on it, so let's move on.